### ISSUE TITLE
[Desktop Mode] Using the scroll wheel to change walking speed

### ISSUE THREAD

**Issue by Khosumi** [Neos-Metaverse/NeosPublic#2182]
When in desktop mode, Your speed is restricted to only sprinting, slow walk and run.

## Describe the solution you'd like
Using the mouse scroll wheel, it would be possible to choose a speed with more control over how fast your character moves.
User chosen thresholds for maximum and minimum speed would also be beneficial for this.

**Comment by Khosumi** [comment]
> I don't think using the scroll wheel would be a good design choice as scrolling UIs is used by the scroll wheel, and it would result in confusion when it does not behave as expected.

That would be simply solved by preventing the scroll wheel from changing movement speed when using UIs.
Scroll wheel feels very natural to change speed, in an analog fashion just like the joysticks on a controller. 
Moreover, tapping the W key will only result in it overriding the walk toggle, so not a solution.

**Comment by 3x1t-5tyl3** [comment]
Currently the UI's allow you to scroll from whereever. Ergo if you were to do scroll to change speed, wouldn't that introduce a max distance for any scrolling UI for desktop users? Or that you'd need an override to change your speed the same way that scale exists? 🤔

**Comment by Khosumi** [comment]
As long as your cursor is on a UI, it shouldn't change the speed, no matter how far away.

**Comment by Frooxius** [comment]
Using Scroll wheel definitely isn't a good idea, it's already used for several other actions so doing this would be very confusing from UX perspective and difficult to do in realtime as well. With keyboard and mouse you simply don't have analog inputs as you have with VR controllers or Gamepad, so emulating those will always be clunky and not something we're likely to implement officially. 

What Neos does with keys already isn't too unusual from plethora of other first and third person games and platforms. Unless there's some intuitive way to implement this without problematic UX that can confuse the user, I don't think it's something we'd implement and it might be better to wait on the Input Binding customization so you can configure this for yourself if you really want it.

**Comment by Khosumi** [comment]
By UX you mean user experience, right? Because I disagree that it'd be confusing.

Star Citizen does it perfectly https://youtu.be/xwLAHcO9nMM?t=161

**Comment by shiftyscales** [comment]
> 
> 
> By UX you mean user experience, right? Because I disagree that it'd be confusing.
> 
> Star Citizen does it perfectly https://youtu.be/xwLAHcO9nMM?t=161

But Star Citizen isn't Neos, and doesn't utilize the scroll wheel in the other sorts of ways that we do.

As Frooxius noted, this seems like it'd be something better suited to a custom configuration a user sets for themself.

**Comment by Khosumi** [comment]
![image](https://user-images.githubusercontent.com/26034954/117350246-588fe300-ae7a-11eb-85ed-950320363644.png)
I have created a working proof of concept. It uses the mouse wheel delta to detect a positive or negative value change and changes the speed value according to a user specified float, to have granular control of how much each wheel tick affects the final speed change.

**Comment by Frooxius** [comment]
Yes, Star Citizen isn't Neos and doesn't have the same interactions as we do. Having the speed control randomly toggle on and off as you hover over interfaces would be problematic, as well as just having too many different functions overloaded on a single control.

Generally with UX you don't want to have too many context-specific different functions on the same input, as it tends to confuse users.
